pkg/compact/compact.go

```diff
@@ -221,23 +227,42 @@ func (s *Syncer) GarbageCollect(ctx context.Context) error {
 
 	begin := time.Now()
 
-	garbageIds := s.duplicateBlocksFilter.DuplicateIDs()
-	for _, id := range garbageIds {
+	duplicateIDs := s.duplicateBlocksFilter.DuplicateIDs()
+	deletionMarkIDs := s.ignoreDeletionMarkFilter.DeletionMarkBlocks(0)
```

Comment:
What about returning map there? (:

Reply:
I am returning a map now, marking this thread as resolved